simpleviewinc/keg-cli
# Write injected compose

## Context
* There are some values required in the docker-compose files that are keg-cli spacific
* Injected apps require these values, but should not have to know about them

## Goal

* Make writing docker-compose files more simple for injected apps
* Fix a number of bugs

## Updates

* Updated all containers to use `/bin/bash`
  * Makes things more consistent
* Added dynamic docker-compose.yml file for injected apps
  * This makes writing compose files for injected apps much more simple
  * Creates a .tmp folder and stores the temp compose file
    * When service is kill, the temp compose file is removed
* Added a runService
  * Runs a docker image, overwriting the default entry point
  * Uses docker directly, not docker-compose
  * Loads volumes / ports form docker-compose.yml file if it exists 
  * Syncs the local directory with the docker container
* Added parsing of the default docker-compose.yml file
  * Allows pulling in the serivce name, ports and volumes
  * Used in the runService
* Removed a number if methods no longer being used
* Cleaned up a few number of files
* Fixed issue with using the --install option when running an app
  * Added creating the `.npmrc` in the containers `run.sh` 
    * This is for internal apps only. Not for injected apps

## **IMPORTANT**
  * Past docker packages will fail due to the `sh` to `bash` change
  * Future docker packages will work as expected

## Testing

* Pull down this branch
  * Run `keg cli update`
  * Check your `~/.kegConfig/cli.config.json` file
    * There should now be a `kegConfig` key in the `paths` object
  * Check your `~/.kegConfig/defaults.env` file
    * At the bottom there should now be a `KEG_CONFIG_PATH` ENV pointing to `{{ cli.paths.kegConfig }}`
  
* Run `keg base build`
  * Should build the base image
* Run `keg base run`
  * Should run the base image
  * Should sync keg-cli with docker container
  * Ensure sync exists with `keg sy`
* Run `keg base rm`
  * Should remove the base image
  * Should remove the base sync
  * Ensure syncs are removed with `keg sy`
* Run `keg core start`
  * Should build the base image
  * Should build and run keg-core
* Run `keg core kill`
  * Should stop keg-core container, and remove all syncs
  * Ensure syncs are removed with `keg sy`
* Run `keg components start`
  * Should build and run keg-components
* Run `keg components kill`
  * Should stop keg-components container, and remove all syncs
  * Ensure syncs are removed with `keg sy`
* Clone this repo => [Keg-External-Example](https://github.com/simpleviewinc/keg-external-example)
  * Follow the instructions in the `README.md`
